app/models/miq_expression/field.rb

```diff
@@ -2,7 +2,7 @@ class MiqExpression::Field
   FIELD_REGEX = /
 (?<model_name>([[:upper:]][[:alnum:]]*(::)?)+)
 \.?(?<associations>[a-z_\.]+)*
--(?<column>[a-z]+(_[a-z]+)*)
+-(?<column>[[a-z]]+(_[[:alnum:]]+)*)
```

Comment:
@imtayadeway I also added elimination of part of hash with 'find', custom attributes cannot be here and Field.parse() was failing on some specialities like '<count>' in 'find' hashes:

```
MiqReport.find_by(:name => 'VMs Snapshot Summary').conditions.exp
  MiqReport Load (0.5ms)  SELECT  "miq_reports".* FROM "miq_reports" WHERE "miq_reports"."name" = $1 LIMIT $2  [["name", "VMs Snapshot Summary"], ["LIMIT", 1]]
  MiqReport Inst Including Associations (0.2ms - 1rows)
{
    "and" => [
        [0] {
            "IS NOT NULL" => {
                "field" => "Vm.snapshots-name",
                "value" => ""
            }
        },
        [1] {
            "FIND" => {
                    "search" => {
                    "IS NOT EMPTY" => {
                        "field" => "Vm.snapshots-name"
                    }
                },
                "checkcount" => {
                    ">" => {
                        "field" => "<count>",
                        "value" => 0
                    }
                }
            }
        }
    ]
}
```